### ISSUE TITLE
elfio doesn't seem to be able to read notes from coredump

### ISSUE THREAD

**Issue by isage** [serge1/ELFIO#93]
While `readelf -n ` and `objdump -h` works, elfio doesn't seem to be able to auto-parse (or, rather, auto-create) note sections from note segments.

Right now i'm using this workaround:
```
ELFIO::Elf_Half n = reader.segments.size();

        for ( ELFIO::Elf_Half i = 0; i < n; ++i )
        {
            ELFIO::segment* seg = reader.segments[i];
            if (seg->get_type() == PT_NOTE)
            {
                std::ostringstream name;
                name << "note" << (int)i;
                ELFIO::section* note_sec = reader.sections.add(name.str());
                note_sec->set_type( SHT_NOTE );
                note_sec->set_data( seg->get_data(), seg->get_file_size() );
            }
        }
```

**Comment by isage** [comment]
Here's sample:
https://transfer.sh/8Axwv9/psp2core-1255533211-0x0000072aa7-eboot.bin.psp2dmp

my sample is a workaround, because elfio fails to see sections in that file, only segments.

**Comment by serge1** [comment]
> Here's sample: https://transfer.sh/8Axwv9/psp2core-1255533211-0x0000072aa7-eboot.bin.psp2dmp

For some reason, access to this file is not available: "This site can’t be reached"

**Comment by isage** [comment]
https://drive.google.com/file/d/101RicL-nzIsaloaD4fxBWjoOrRUwjD2A/view?usp=sharing
how about this?

**Comment by serge1** [comment]
Thank you for the example. I see what you mean now.
I am trying to extend note section accessor to be able to work with segments as well.

**Comment by serge1** [comment]
Hi,
Note segment accessor is available in commit feee9d4.
In addition, elfdump prints out this data too.

**Comment by isage** [comment]
Awesome, thank you!
